autopush-common/src/db/dual/mod.rs

```diff
@@ -0,0 +1,288 @@
+        let presult = self.primary.remove_channel(uaid, channel_id).await?;
+        let sresult = self.secondary.remove_channel(uaid, channel_id).await?;
+        Ok(presult | sresult)
+    }
+
+    async fn remove_node_id(
+        &self,
+        uaid: &Uuid,
+        node_id: &str,
+        connected_at: u64,
+    ) -> DbResult<bool> {
+        let p_resp = self
+            .secondary
+            .remove_node_id(uaid, node_id, connected_at)
+            .await?;
+        let s_resp = self
+            .primary
+            .remove_node_id(uaid, node_id, connected_at)
+            .await?;
+        return Ok(p_resp || s_resp);
+    }
+
+    async fn save_message(&self, uaid: &Uuid, message: Notification) -> DbResult<()> {
+        if self.write_to_secondary {
+            let _ = self.secondary.save_message(uaid, message.clone()).await?;
+        }
+        self.primary.save_message(uaid, message).await
+    }
+
+    async fn remove_message(&self, uaid: &Uuid, sort_key: &str) -> DbResult<()> {
+        let _ = self.secondary.remove_message(uaid, sort_key).await;
+        self.primary.remove_message(uaid, sort_key).await
+    }
+
+    async fn fetch_topic_messages(
+        &self,
+        uaid: &Uuid,
+        limit: usize,
+    ) -> DbResult<FetchMessageResponse> {
+        let result = self.primary.fetch_topic_messages(uaid, limit).await?;
+        if result.messages.is_empty() {
+            return self.secondary.fetch_topic_messages(uaid, limit).await;
+        }
+        return Ok(result);
+    }
+
+    async fn fetch_timestamp_messages(
+        &self,
+        uaid: &Uuid,
+        timestamp: Option<u64>,
+        limit: usize,
+    ) -> DbResult<FetchMessageResponse> {
+        let result = self
+            .primary
+            .fetch_timestamp_messages(uaid, timestamp, limit)
+            .await?;
+        if result.messages.is_empty() {
+            return self
+                .secondary
+                .fetch_timestamp_messages(uaid, timestamp, limit)
+                .await;
+        }
+        return Ok(result);
+    }
+
+    async fn save_messages(&self, uaid: &Uuid, messages: Vec<Notification>) -> DbResult<()> {
+        if self.write_to_secondary {
+            let _ = self.secondary.save_messages(uaid, messages.clone()).await?;
+        }
+        self.primary.save_messages(uaid, messages).await
+    }
+
+    async fn increment_storage(&self, uaid: &Uuid, timestamp: u64) -> DbResult<()> {
+        if self.write_to_secondary {
+            let _ = self.secondary.increment_storage(uaid, timestamp).await?;
+        }
+        self.primary.increment_storage(uaid, timestamp).await
+    }
+
+    async fn health_check(&self) -> DbResult<bool> {
+        Ok(self.primary.health_check().await? && self.secondary.health_check().await?)
+    }
+
+    async fn router_table_exists(&self) -> DbResult<bool> {
+        self.primary.router_table_exists().await
```

Comment:
I was debating that. 
I wanted the call to be reasonably fast and working through the logic around which sources to poll might be weird. Plus, since we don't pass the UAID, we're kinda arbitrarily picking which data store to read from anyway. 

Picking just primary ensures that the prime database is correct, with the secondary being just that, and lesser priority.  (Honestly, once we're on bigtable, we could drop this entirely, but there's probably good reasons to keep it in case we ever have to move to another data store).